e2e/nomostest/testpredicates/predicates.go

```diff
@@ -456,38 +457,35 @@ func NoConfigSyncMetadata() Predicate {
 	}
 }
 
-// AllResourcesAreCurrent ensures that the managed resources
-// are all Current in the ResourceGroup CR.
-func AllResourcesAreCurrent() Predicate {
-	return func(o client.Object) error {
-		if o == nil {
+// AllResourcesReconciled ensures that the managed resources
+// are all reconciled in the ResourceGroup CR.
+func AllResourcesReconciled(scheme *runtime.Scheme) Predicate {
+	return func(obj client.Object) error {
+		if obj == nil {
 			return ErrObjectNotFound
 		}
-		u, ok := o.(*unstructured.Unstructured)
-		if !ok {
-			return WrongTypeErr(u, &unstructured.Unstructured{})
-		}
-		resourceStatuses, found, err := unstructured.NestedSlice(u.Object, "status", "resourceStatuses")
-		if err != nil {
-			return err
-		}
-		if !found {
-			return fmt.Errorf("resource status not found in %v", u)
-		}
-		for _, resource := range resourceStatuses {
-			s, ok := resource.(map[string]interface{})
-			if !ok {
-				return WrongTypeErr(s, map[string]interface{}{})
+		if uObj, ok := obj.(*unstructured.Unstructured); ok {
+			rObj, err := kinds.ToTypedObject(uObj, scheme)
+			if err != nil {
+				return err
 			}
-			status, found, err := unstructured.NestedString(s, "status")
+			obj, err = kinds.ObjectAsClientObject(rObj)
 			if err != nil {
 				return err
 			}
-			if !found {
-				return fmt.Errorf("status field not found for resource %v", resource)
+		}
+		rgObj, ok := obj.(*kptv1alpha1.ResourceGroup)
+		if !ok {
+			return WrongTypeErr(obj, rgObj)
+		}
+		for _, resource := range rgObj.Status.ResourceStatuses {
```

Comment:
What if the status hasn't been updated yet? Is it possible the status is missing objects declared in the spec?

Should we also check `observedGeneration`?

Reply:
Good idea.

Given that I always ran this after checking that the RSync was Synced, I don't think I need to re-run the profile tests.

Reply:
Added.